### DISCUSSION TITLE
REACT 18 Version

### DISCUSSION THREAD

**Discussion by IN2TEC** [react-page/react-page#1308] [General] [answered]
Hi there,

first of all, I would like to thank all contributors for this excellent component. We would love to integrate it with our site. Unfortunately, we are using REACt 18 with Mui5 and not getting it running. We saw the note on the documentation that it is not compatible and the discussion on GutHub in regards to the open issues. But I wonder if and when there will be a new release and/or what it takes to get it done. It would be a shame not to keep this package going.

Best
Mat

**Comment by macrozone** [comment, marked as answer]
it works with react 18 and mui 5.

if not, best follow the documentation and check if you do everything properly https://react-page.github.io/docs/#/quick-start

then check whether use the latest versions.

**Reply by IN2TEC** [reply]
Thank you for your feefback. Unfortunately it doesn't work for me, even not with a fresh install. The react-dnd module throws errors stating it the jsxruntime is not exporter. 

I'll try again and post screenshots later

**Comment by IN2TEC** [comment]
Good Morning @macrozone, sorry for the delay of my screenshot with the error message. For better support, I created a public repository with my starter code which I created as of this morning by following the following documentation

- Getting Started "https://react-page.github.io/docs/#/quick-start" as well as the 
- Mui Migration https://mui.com/x/migration/migration-data-grid-v4/#using-mui-core-v4-with-v5
- Simple Example of Editor Component https://react-page.github.io/docs/#/editor?id=intro

Is there any way you can help me out on this?  I would really like to use your library for our blog but kind of desperate in getting it working, because I spent a few days trying to debug it?

Here is the link to our public repo with the code I produced this morning: https://github.com/IN2TEC/react-playground

![Screenshot 2023-09-28 at 3 36 56 AM](https://github.com/react-page/react-page/assets/104642539/4e4a63b3-108e-4fac-a9e2-a0dceb858703)

**Comment by IN2TEC** [comment]
Hey there, i think I found the issue, which is REACT DND. I changed the code to the latest (according to REACT documentation) abnd voila! 

Look at 
- import { jsx as _jsx } from "react/jsx-runtime.js"; which i replaced with import React from 'react';
- return /*#__PURE__*/ _jsx(DndContext.Provider..... which i replaced with return /*#__PURE__*/ React.createElement(DndContext.Provider....

Edited File node_modules/react-dnd/dist/esm/core/DndProvider.mjs

```
function _objectWithoutProperties(source, excluded) {
    if (source == null) return {};
    var target = _objectWithoutPropertiesLoose(source, excluded);
    var key, i;
    if (Object.getOwnPropertySymbols) {
        var sourceSymbolKeys = Object.getOwnPropertySymbols(source);
        for(i = 0; i < sourceSymbolKeys.length; i++){
            key = sourceSymbolKeys[i];
            if (excluded.indexOf(key) >= 0) continue;
            if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue;
            target[key] = source[key];
        }
    }
    return target;
}
function _objectWithoutPropertiesLoose(source, excluded) {
    if (source == null) return {};
    var target = {};
    var sourceKeys = Object.keys(source);
    var key, i;
    for(i = 0; i < sourceKeys.length; i++){
        key = sourceKeys[i];
        if (excluded.indexOf(key) >= 0) continue;
        target[key] = source[key];
    }
    return target;
}
// import { jsx as _jsx } from "react/jsx-runtime.js";
import React from 'react';
import { useEffect, memo } from 'react';
import { createDragDropManager } from 'dnd-core';
import { DndContext } from './DndContext.mjs';
let refCount = 0;
const INSTANCE_SYM = Symbol.for('__REACT_DND_CONTEXT_INSTANCE__');
var DndProvider = /*#__PURE__*/ memo(function DndProvider(_param) {
    var { children  } = _param, props = _objectWithoutProperties(_param, [
        "children"
    ]);
    const [manager, isGlobalInstance] = getDndContextValue(props) // memoized from props
    ;
    /**
		 * If the global context was used to store the DND context
		 * then where theres no more references to it we should
		 * clean it up to avoid memory leaks
		 */ useEffect(()=>{
        if (isGlobalInstance) {
            const context = getGlobalContext();
            ++refCount;
            return ()=>{
                if (--refCount === 0) {
                    context[INSTANCE_SYM] = null;
                }
            };
        }
        return;
    }, []);
    return /*#__PURE__*/ React.createElement(DndContext.Provider, {
        value: manager,
        children: children
    });
});
/**
 * A React component that provides the React-DnD context
 */ export { DndProvider,  };
function getDndContextValue(props) {
    if ('manager' in props) {
        const manager = {
            dragDropManager: props.manager
        };
        return [
            manager,
            false
        ];
    }
    const manager = createSingletonDndContext(props.backend, props.context, props.options, props.debugMode);
    const isGlobalInstance = !props.context;
    return [
        manager,
        isGlobalInstance
    ];
}
function createSingletonDndContext(backend, context = getGlobalContext(), options, debugMode) {
    const ctx = context;
    if (!ctx[INSTANCE_SYM]) {
        ctx[INSTANCE_SYM] = {
            dragDropManager: createDragDropManager(backend, context, options, debugMode)
        };
    }
    return ctx[INSTANCE_SYM];
}
function getGlobalContext() {
    return typeof global !== 'undefined' ? global : window;
}

//# sourceMappingURL=DndProvider.mjs.map
```

I blieve it is a react-dnd problem as they have the same code in their latest version. Is there anyway I can ovverride there module?
